sound/soc/sof/intel/apl.c

```diff
@@ -40,6 +40,110 @@ static const struct snd_sof_debugfs_map apl_dsp_debugfs[] = {
 	{"dsp", HDA_DSP_BAR,  0, 0x10000},
 };
 
+static const struct snd_soc_dai_ops apl_dai_ops = {
+};
+
+#define APL_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S24_LE | \
+	SNDRV_PCM_FMTBIT_S32_LE | SNDRV_PCM_FMTBIT_FLOAT)
+
+/* Apollolake DAIs */
+static struct snd_soc_dai_driver apl_dai[] = {
+{
+	.name = "SSP0 Pin",
+	.ops = &apl_dai_ops,
+	.playback = SOF_DAI_STREAM("ssp0 Tx", 1, 16,
```

Comment:
it original from Liam, And it is the maximum of cpu_dai caps, it can still be reduced by the FE/PCM caps:
	hw->channels_min = max(chan_min, cpu_stream->channels_min);
	hw->channels_max = min(chan_max, cpu_stream->channels_max);

so I think leaving them as 16 has no harm.

Reply:
8 chan is correct, my original value is wrong.